src/C++/Utility.cpp

```diff
@@ -320,6 +320,9 @@ const char* socket_hostname(socket_handle socket )
 
 const char* socket_hostname( const char* name )
 {
+
+  if ( name == 0 ) return 0;
+
   struct hostent* host_ptr = 0;
```

Comment:
Previously did not have a check for this as a new test ended up with an exception